pkg/chunk/chunk_store.go

```diff
@@ -482,3 +487,126 @@ func (c *store) convertChunkIDsToChunks(ctx context.Context, userID string, chun
 
 	return chunkSet, nil
 }
+
+func (c *store) DeleteChunk(ctx context.Context, from, through model.Time, userID, chunkID string, metric labels.Labels, partiallyDeletedInterval *model.Interval) error {
```

Comment:
Now that I'm getting a better understand of the logic, I think that passing `partiallyDeletedInterval` is a bit confusing. Have you considered splitting the interface into:
- `DeleteChunk(ctx context.Context, from, through model.Time, userID, chunkID string, metric labels.Labels)`
- `ReboundChunk(ctx context.Context, from, through model.Time, userID, chunkID string, metric labels.Labels, reboundInterval model.Interval)`

Internally, it's perfectly fine to have the same shared implementation, but splitting the interface would make the intention more clear in my opinion:
- `DeleteChunk()` deletes an entire chunk
- `ReboundChunk()` deletes a portion of the chunk, rebounding it

I would like to know what's your take on this.

Reply:
_Let's keep this as food for thought for the next PR. We can eventually iteratively address it._